File: src/ros2_bridge/mappers/geometry_msgs/accel_with_covariance.rs
//! Typed mapper for `geometry_msgs/msg/AccelWithCovariance`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn accel_with_covariance_to_bus(msg: ros_env::geometry_msgs::msg::AccelWithCovariance) -> crate::geometry_msgs::msg::v1::AccelWithCovariance {
    crate::geometry_msgs::msg::v1::AccelWithCovariance {
        accel: Some(crate::ros2_bridge::mappers::geometry_msgs::accel::accel_to_bus(msg.accel)),
        covariance: crate::ros2_bridge::mappers::convert::f64_seq(msg.covariance),
    }
}

pub(crate) fn accel_with_covariance_to_ros(bus: crate::geometry_msgs::msg::v1::AccelWithCovariance) -> ros_env::geometry_msgs::msg::AccelWithCovariance {
    ros_env::geometry_msgs::msg::AccelWithCovariance {
        accel: crate::ros2_bridge::mappers::geometry_msgs::accel::accel_to_ros(bus.accel.unwrap_or_default()),
        covariance: crate::ros2_bridge::mappers::convert::FromF64Seq::from_f64_seq(bus.covariance),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct GeometryMsgsAccelWithCovarianceMapper;

impl TypedTopicMapper for GeometryMsgsAccelWithCovarianceMapper {
    type Ros = ros_env::geometry_msgs::msg::AccelWithCovariance;
    type Bus = crate::geometry_msgs::msg::v1::AccelWithCovariance;

    fn type_name(&self) -> &'static str {
        "geometry_msgs/msg/AccelWithCovariance"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(accel_with_covariance_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(accel_with_covariance_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/geometry_msgs/accel_with_covariance.py
"""Generated mapper for `geometry_msgs/msg/AccelWithCovariance`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.geometry_msgs.accel import accel_to_bus, accel_to_ros

def accel_with_covariance_to_bus(msg):
    from robot_bus.geometry_msgs.msg.v1 import AccelWithCovariance as BusMsg

    bus = BusMsg()
    bus.accel.CopyFrom(accel_to_bus(msg.accel))
    bus.covariance.extend(list(msg.covariance))
    return bus


def accel_with_covariance_to_ros(bus):
    from geometry_msgs.msg import AccelWithCovariance as RosMsg

    out = RosMsg()
    out.accel = accel_to_ros(bus.accel)
    out.covariance = list(bus.covariance)
    return out


class GeometryMsgsAccelWithCovarianceMapper:
    def type_name(self) -> str:
        return "geometry_msgs/msg/AccelWithCovariance"

    def ros_msg_type(self):
        from geometry_msgs.msg import AccelWithCovariance as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return accel_with_covariance_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.geometry_msgs.msg.v1 import AccelWithCovariance as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return accel_with_covariance_to_ros(bus)
